operator/internal/controller/vectorized/cluster_controller.go

```diff
@@ -510,52 +506,85 @@ func (r *ClusterReconciler) reportStatus(
 	nodeList.Internal = observedNodesInternal
 	nodeList.SchemaRegistry.Internal = fmt.Sprintf("%s:%d", clusterFQDN, schemaRegistryPort)
 
-	//nolint:nestif // the code won't get clearer if it's splitted out in my opinion
-	version, versionErr := sts.CurrentVersion(ctx)
-	if versionErr != nil {
-		// this is non-fatal error, it will return error even if e.g.
-		// the rollout is not finished because then the currentversion
-		// of the cluster cannot be determined
-		r.Log.Info(fmt.Sprintf("cannot get CurrentVersion of statefulset, %s", versionErr))
+	if len(stSets) == 0 {
+		r.Log.Info("no stateful sets found")
 	}
-	if statusShouldBeUpdated(&redpandaCluster.Status, nodeList, sts, version, versionErr) {
-		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			cluster := &vectorizedv1alpha1.Cluster{}
-			err := r.Get(ctx, types.NamespacedName{
-				Name:      redpandaCluster.Name,
-				Namespace: redpandaCluster.Namespace,
-			}, cluster)
-			if err != nil {
-				return err
-			}
 
-			cluster.Status.Nodes = *nodeList
-			cluster.Status.ReadyReplicas = sts.LastObservedState.Status.ReadyReplicas
-			cluster.Status.Replicas = sts.LastObservedState.Status.Replicas
-			if versionErr == nil {
-				cluster.Status.Version = version
-			}
+	var version string
+	var versionErr error
 
-			err = r.Status().Update(ctx, cluster)
-			if err == nil {
-				// sync original cluster variable to avoid conflicts on subsequent operations
-				*redpandaCluster = *cluster
-			}
-			return err
-		})
+	nodePoolStatus := make(map[string]vectorizedv1alpha1.NodePoolStatus)
+	readyReplicas := int32(0)
+	replicas := int32(0)
+	currentReplicas := int32(0)
+	for _, sts := range stSets {
+		if sts.LastObservedState == nil {
+			return errNonexistentLastObservedState
+		}
+
+		readyReplicas += sts.LastObservedState.Status.ReadyReplicas
+		replicas += sts.LastObservedState.Status.Replicas
+
+		oldNps := redpandaCluster.Status.NodePools[sts.GetNodePool().Name]
```

Comment:
i need to double check if this is correct. even if there is no status yet for this NP, we do want to create one. the existing code will just do that, as it will have a status struct with default values returned